dfir_rs/src/util/priority_stack.rs

```diff
@@ -0,0 +1,83 @@
+//! A priority queue in which elements of the same priority are popped in a LIFO order.
+
+/// A priority stack in which elements of the same priority are popped in a LIFO order.
+#[derive(Debug, Clone)]
+pub struct PriorityStack<T> {
+    /// Note: inner stack `Vec`s may be empty.
+    stacks: Vec<Vec<T>>,
+}
+
+impl<T> PriorityStack<T> {
+    /// Creates a new, empty `PriorityStack`.
+    pub fn new() -> Self {
+        Self {
+            stacks: Vec::default(),
+        }
+    }
+
+    /// Pushes an element onto the stack with the given priority.
+    pub fn push(&mut self, priority: usize, item: T) {
+        if priority >= self.stacks.len() {
+            self.stacks.resize_with(priority + 1, Default::default);
```

Comment:
is this inefficient to do one at a time?

Reply:
Only needs to be initialized once, the priorities are never shrunk. Though we could to this earlier